### PR TITLE
feat(metric_alerts): Make metric alerts work in local dev.

### PR DESCRIPTION
**Pull request by wedamija**
This makes the subscription consumer run if `SENTRY_DEV_PROCESS_SUBSCRIPTIONS` is true. Also starts
Kafka and fixes some bugs around auto topic creation that we don't see in prod.

We still need a separate pr in snuba to run the consumer there.

### CODE REVIEW

**Comment by dashed** [comment]
Is it safe to do this without setting this env var explicitly? 

```
SENTRY_EVENTSTREAM = "sentry.eventstream.kafka.KafkaEventStream"
```

**Comment by wedamija** [comment]
> Is it safe to do this without setting this env var explicitly?
> 
> ```
> SENTRY_EVENTSTREAM = "sentry.eventstream.kafka.KafkaEventStream"
> ```

Yeah, all we're doing here is starting up Kafka so it's available for use with subscriptions. We use completely separate topics, so it should be fine.

**Comment by wedamija** [comment]
> > Is it safe to do this without setting this env var explicitly?
> > ```
> > SENTRY_EVENTSTREAM = "sentry.eventstream.kafka.KafkaEventStream"
> > ```
> 
> Yeah, all we're doing here is starting up Kafka so it's available for use with subscriptions. We use completely separate topics, so it should be fine.

Damn, I just realised that this can only work if we're using `KafkaEventStream`. Otherwise, `snuba devservices` will run with `--no-workers`. I guess I'll add an error here if people try to run subscriptions with the snuba event stream.

File: src/sentry/runner/commands/devserver.py [hunk 1]
```diff
@@ -21,11 +21,22 @@
...
+    ],
 }
 
 
-def _get_daemon(name):
-    return (name, _DEFAULT_DAEMONS[name])
+def _get_daemon(name, *args, **kwargs):
+    display_name = name
+    if "suffix" in kwargs:
+        display_name = "{}-{}".format(name, kwargs["suffix"])
```
Comment (EvanPurkhiser) [inline src/sentry/runner/commands/devserver.py:37]: prefix with `u`

**Review by evanpurkhiser — APPROVED** [review]

**Comment by wedamija** [comment]
Just noting this works ok, but if you don't run your server for a while then the next time you run it then it'll process the queries for every minute since you last ran, which isn't really practical for dev. I'll fix this in snuba, will probably just skip to latest offset when we start the subscription processors in dev mode.

**Comment by scefali** [comment]
@wedamija I tried getting this to work by adding the following to my `sentry.conf.py` file:
```
SENTRY_DEV_PROCESS_SUBSCRIPTIONS = True
SENTRY_EVENTSTREAM="sentry.eventstream.kafka.KafkaEventStream"
```
But I get this error:
```
Exception: KafkaError{code=UNKNOWN_TOPIC_OR_PART,val=3,str="Subscribed topic not available: events: Broker: Unknown topic or partition"}
```
Any idea what's going on?

**Comment by wedamija** [comment]
> @wedamija I tried getting this to work by adding the following to my `sentry.conf.py` file:
> 
> ```
> SENTRY_DEV_PROCESS_SUBSCRIPTIONS = True
> SENTRY_EVENTSTREAM="sentry.eventstream.kafka.KafkaEventStream"
> ```
> 
> But I get this error:
> 
> ```
> Exception: KafkaError{code=UNKNOWN_TOPIC_OR_PART,val=3,str="Subscribed topic not available: events: Broker: Unknown topic or partition"}
> ```
> 
> Any idea what's going on?

Just want to check: Have you pulled latest snuba? `sentry devservices down`, `sentry devservices up`.

After some more testing locally, there are a couple of other things I need to do to make this work well, but it should generally work at the moment.

**Comment by scefali** [comment]
>Just want to check: Have you pulled latest snuba? sentry devservices down, sentry devservices up.

@wedamija good call, that worked!

**Pull request merged**